### PR TITLE
DAH-2876 feat: allow redirect after sign in

### PR DESCRIPTION
**Pull request by cade-exygy**
## Description

If a user tries to go to the my-account, account-settings, or my-applications and is not signed in: redirect to sign in page, show banner. Once the login happens, the user is redirected to the page they were trying to access prior.

## Jira ticket

https://sfgovdt.jira.com/browse/DAH-2876

### Version Control

- [x] branch name begins with `angular` if it contains updates to Angular code
- [x] branch name contains the Jira ticket number
- [x] PR name follows `type: TICKET-NUMBER Description` format, e.g. `feat: DAH-123 New Feature`

### Code quality

- [x] [the set of changes is small](https://google.github.io/eng-practices/review/developer/small-cls.html#what-is-small)
- [ ] all automated code checks pass (linting, tests, coverage, etc.)
- [x] code irrelevant to the ticket is not modified e.g. changing indentation due to automated formatting
- [x] if the code changes the UI, it matches the UI design exactly
- [x] if the changes include human translations, follow the [human translations process](https://sfgovdt.jira.com/l/cp/XS1KpvE4)

### Review instructions

- [x] instructions specify which environment(s) it applies to
- [x] instructions work for PA testers
- [x] instructions have already been performed at least once

### Request review

- [x] PR has `needs review` label
- [x] Use `Housing Eng` group to automatically assign reviewers, and/or assign specific engineers
- [x] If time sensitive, notify engineers in Slack

## Review Instructions
- On the review app:
- While signed out, manually enter the restricted url: /my-account?react=true
- It should try to load the page, but then redirect back to /sign-in
- You should see the banner
- Sign In to an account: test@test.com / test1234
- Once you are signed in, you should be redirected
- Try this with /my-account, /account-settings, and /my-applications

Test edge cases like signing out, triggering other banners, etc

### CODE REVIEW

**Comment by tallulahkay** [comment]
@cade-exygy could you try to up the code cov here?

File: app/javascript/util/routeUtil.ts [hunk 1]
```diff
@@ -75,3 +75,22 @@ export const getEligibilityEstimatorLink = localizedPathGetter("/eligibility-est
...
+
+const SignInRedirects = {
+  account: getMyAccountPath(),
+  applications: getMyApplicationsPath(),
+  settings: getMyAccountSettingsPath(),
+  home: getHomepagePath(),
+}
+
+const getRedirectUrl = (key: string): string => {
+  console.log("getting redirect for", key)
```
Comment (jimlin-sfgov) [inline app/javascript/util/routeUtil.ts:87]: issue: Remove debugging line.

File: app/javascript/pages/account/account-settings.tsx [hunk 2]
```diff
@@ -456,7 +457,8 @@ const AccountSettingsPage = () => {
   const { profile, loading, initialStateLoaded } = React.useContext(UserContext)
 
   if (!profile && !loading && initialStateLoaded) {
-    // TODO: Redirect to React sign in page and show a message that user needs to sign in
+    window.sessionStorage.setItem("redirect", "settings")
```
Comment (jimlin-sfgov) [inline app/javascript/pages/account/account-settings.tsx:460]: thought: We're using `sessionStorage` across various files, and it's not immediately obvious what arguments for `setItem` mean. We should look into consolidating all routing-related `sessionStorage` usage into a single file, probably in `routeUtil.ts`.

File: app/javascript/pages/account/my-account.tsx [hunk 3]
```diff
@@ -62,7 +63,8 @@ const MyAccount = (_props: MyAccountProps) => {
   const { profile, loading, initialStateLoaded } = React.useContext(UserContext)
 
   if (!profile && !loading && initialStateLoaded) {
-    // TODO: Redirect to React sign in page and show a message that user needs to sign in
+    window.sessionStorage.setItem("redirect", "account")
+    setSiteAlertMessage(t("signIn.loginRequired"), "secondary")
```
Comment (jimlin-sfgov) [inline app/javascript/pages/account/my-account.tsx:67]: thought: We're well along this path now, but in the future we should look into using constants instead of strings for things like "redirect", "account", and "secondary".

**Review by jimlin-sfgov — CHANGES_REQUESTED** [review]
